base/db/R/dbfiles.R

```diff
@@ -251,13 +251,13 @@ dbfile.input.check <- function(siteid, startdate = NULL, enddate = NULL, mimetyp
   formatid <- get.id(table = "formats", colnames = c("mimetype_id", "name"), values = c(mimetypeid, formatname), con = con)
 
   if (is.null(formatid)) {
-    invisible(data.frame())
+    return (invisible(data.frame()))
   }
 
   # setup parent part of query if specified
-  if (is.na(parentid)) {
-    parent <- ""
-  } else {
+  parent <- ""
+  
+  if (!is.na(parentid)) {
     parent <- paste0(" AND parent_id=", parentid)
   }
```

Comment:
👍 This same pattern appears in dbfile.input.insert above  too -- can you modify it there (near line 60) as well?